File: CourseProjects/CourseProjects/VisualStudioProjects/DesignPatternsExamples/TemplateMethod/SmtpConnection.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace TemplateMethod {
    class SmtpConnection : Connection {
        protected override void OpenConnection() {
            Console.WriteLine("SmtpConnection.Open");
        }

        protected override void SendData(char[] data) {
            Console.WriteLine("SmtpConnection.Send");
        }

        protected override void CloseConnection() {
            Console.WriteLine("SmtpConnection.Close");
        }
    }
}

File: CourseProjects/CourseProjects/JavaEclipseProjects/DesignPatternsExamples/src/patterns/templatemethod/SmtpConnection.java
package patterns.templatemethod;

public class SmtpConnection extends Connection {
    void openConnection() {
        System.out.println("SmtpConnection.openConnection");
    }
    void sendData(byte[] data) {
        System.out.println("SmtpConnection.sendData");
    }
    void closeConnection() {
        System.out.println("SmtpConnection.closeConnection");
    }
}
